src/ImageRunModal.jsx

```diff
@@ -429,9 +429,9 @@ export class ImageRunModal extends React.Component {
         // searching for specific image tags
         // instead only rely on manifests query (requires image:tag name)
         if (!RE_CONTAINER_TAG.test(value)) {
-            // If there are registries configured search in them, or if a user searches for `docker.io/cockpit` let
+            // If there are search registries configured, search in them, or if a user searches for `docker.io/cockpit` let
             // podman search in the user specified registry.
-            if (Object.keys(this.props.podmanInfo.registries).length !== 0 || value.includes('/')) {
+            if (this.props.podmanInfo.registries?.search?.length !== 0 || value.includes('/')) {
```

Comment:
This doesn't work, I am afraid. If `registries` doesn't have a `search` field, `registries?.search` is `undefined`, which is `!== 0`. But we want the condition to be false in that case.

```
>> x = { }
Object {  }
>> x?.search?.length
undefined
>> x?.search?.length !== 0
true 
```

Maybe `registries?.search?.length > 0`?

Reply:
The return of the son of Perl! :-) What about `registries.search && registries.search.length > 0`?

Reply:
Can do as well -- I was mostly trying to avoid having to spell out that long name twice, but I'll do an alias.